### ISSUE TITLE
Extract the integrated terminal

### ISSUE THREAD

**Issue by lloeki** [Microsoft/vscode#34442]
"Feature" request.

The integrated terminal grows in functionality, to the point that it could soon become very useful standalone, as its *own app*. I'm only a casual VScode user (using vim most of the time), but I would definitely be interested in a well-behaved, consistent, reactively-configurable, multi-platform terminal not tied to the editor.

Obviously if such a situation were to unfold, the shared component (not entirely dissimilar to VTE) should be jointly developed and shared with VScode. Both could be able to leverage a common configuration so that whether a terminal leveraging such a theoretical common component is spawned within VScode or standalone, they behave the same.

**Comment by lloeki** [comment]
Hint, hint, call the thing VSterm?

**Comment by Tyriar** [comment]
This is an idea I've thrown around a bit, I doubt we'll get time to work on it any time soon. The first steps for this would be more flexible movement of the panel inside the workbench and pulling it into its own window.

**Comment by auchenberg** [comment]
You should check out https://hyper.is/, also Electron based and extensible.

**Comment by lloeki** [comment]
@auchenberg I definitely know about Hyper ;) yet Code's implementation is currently much more solid though, and given each one's track record in stability and performance I'd rather bet on Code. Also having the external terminal and Code's integrated one behave in exactly the same way is a non-negligible feature.

**Comment by auchenberg** [comment]
@lloeki Hyper is migrating to xterm.js build by @Tyriar, which means that Hyper most likely will be similar to the VS Code console. I can't speak on behalf of @Tyriar, but to me it sounds like most of the concerns are addressed by this migration. https://github.com/zeit/hyper/issues/1275

**Comment by Tyriar** [comment]
Yes Hyper is planning on moving to xterm.js for v2. There will be some differences such as configuration and some things are handled differently (clear in Hyper may not work on Windows for example), overall it will be very close to VS Code though.

**Comment by warpdesign** [comment]
I can't speak for v2 but the last time I tried Hyper at was really slow & buggy, especially under Windows.

VSCode's terminal implementation worked so good that I don't use cmd.exe anymore.

I understand it's not the priority but if I was to try to extract VSCode's terminal and make it run without VSCode where should I start?

**Comment by Tyriar** [comment]
@warpdesign what I was thinking was to be able to launch the terminal by itself using `code --term` or something similar, this would involve:

- Create the "terminal window" which has access to the services used internally by the regular VS Code window (this part would be particularly tricky)
- Modifying the CLI to support this
- Figure out how keybindings and other user settings interact with this special window

Before starting any of this it would probably be wise to get buy in from the team though (and it's not a priority yet so I wouldn't do this yet).

**Comment by nico-onmap** [comment]
@Tyriar Ok. I'll won't do anything yet then.

**Comment by warpdesign** [comment]
@Tyriar Ok. I'll wait then.

**Comment by donaldpipowitch** [comment]
> I can't speak for v2 but the last time I tried Hyper it was really slow & buggy, especially under Windows.

I use the most recent canary and for me it becomes really slow and buggy (disappearing texts, frozen windows and so on) as well, while the terminal in VS Code seems to work really good in everything I tried so far. (Also the copy'n'pasting behavior there seems to work better for me.) Therefor I'd 💕💕💕 to see the VS Code terminal as a stand-alone application as well.

> This is an idea I've thrown around a bit, I doubt we'll get time to work on it any time soon.

Are there small steps we can do to slowly iterate to this point?

**Comment by Tyriar** [comment]
It would be a lot easier if something like this landed in Electron https://github.com/electron/electron/pull/11501, that would allow sharing all the services across multiple windows. Without that, it's hard to think of smaller steps we could take as everything is a pretty big task that might be throw away if the team doesn't like it.

We could try put together a proof of concept in some branch which hacks together the terminal and necessary services, launching via `code --term` or something. This would be useful to get some buy in and to show it's a compelling proposition. I doubt I'll have the time to do this any time soon though.

I'll be moving on with splitting soon which is kind of a prerequisite to this anyway, we wouldn't want a stand alone terminal that couldn't show more than one instance at once. https://github.com/Microsoft/vscode/issues/7504

In the meantime I recommend hooking up a keybinding to toggle maximize on the panel if you haven't yet https://github.com/Tyriar/dotfiles/blob/7cdd4a9838a0cdec4508a1e484fc603f58c9e1c3/modules/vscode/config/keybindings.json#L19, I use it all the time 😃

**Comment by lloeki** [comment]
> This would be useful to get some buy in and to show it's a compelling proposition

This looks like a sound strategy.

> we wouldn't want a stand alone terminal that couldn't show more than one instance at once

Even as a proof of concept, tmux would make this already very useful for me!

**Comment by Tyriar** [comment]
> Even as a proof of concept, tmux would make this already very useful for me!

Still, it's just another thing that will make it even better as this is something on the roadmap that is definitely coming soon 😃

**Comment by bijeebuss** [comment]
I'm glad this is being worked on. The integrated terminal in VS Code is the best terminal on windows and I would love to be able to use it as a standalone app.

**Comment by i0natan** [comment]
+1

**Comment by RobCannon** [comment]
I find that Hyper isn't well supported on Windows.  The integrated term in VS Code is the best implementation that I have found and it would be great to be able to use the same terminal everyone that I use a command line.  Of course, Windows 10 needs a feature to allow the built in terminal to be replaced by a custom terminal.

**Comment by mikemaccana** [comment]
This is a little higher priority for terminal users now that Sets was pulled from Fast Ring Windows 10 builds. 

Sets clearly has a long way to go - starting a new tab was a slow process of opening the new tab, waiting for edge to start (if it didn't crash) and then searching for the app you wanted to start (another powershell, duh). So right now, tabbed terminals is: wait for a bunch of bugs to be fixed, wait for builds to inclode those fixes, wait till you can install a build with those fixes.

Having a **VSTerm** would give us something we know works. 👍🏼

**Comment by huafu** [comment]
#40829 has been marked as a duplicate, but I think they are 2 different concerns.

1. This issue, where people want a VSTerm app or kind of (if I understand well)
2. Issue #40829 where the VSCode integrated terminal panel could be "detached" from the main window.

The benefit of the `#2` is clear when using dual-screen: tests can be run on another screen while keeping the code with another small terminal in the bottom, **WITHOUT loosing the ability to `CMD-click` on failing tests' paths** in the detached terminal running the tests, to directly open their editor in the main VSCode window. I think #40829 should be re-opened as it makes sense it's a different issue from this one.

**Comment by Tyriar** [comment]
@huafu I'm trying to keep the issue count down, there are several issues relating to this and this is how I see them:

- https://github.com/Microsoft/vscode/issues/10121 Pull tabs out into floating windows - This is a blocker to what you're after
- https://github.com/Microsoft/vscode/issues/10546 Tabs for terminal - This would likely be needed to pull *individual* terminals out, not the whole panel (not sure if there's an issue for pulling the panel out or not)

Once both of those are implemented you would naturally get the functionality you're after for free.

**Comment by hgouveia** [comment]
VSTerminal as a standalone app would be excellent 

I've used Hyper.Js and Terminus, both on windows, specially Hyper has a lot of bugs, it didn't work properly for me and it was slow,  VS Terminal works like charm, very configurable , is fast, and packed with some of the Hyper feature

Current Features: 

- Open Multiple Terminal in "tab/dropdown"
- Open Different Terminal (cmd, powershell, bash, etc)
- Split functionality
- Set Custom terminal from settings (terminal, arguments, font, fontsize), just to name a few, i was able to set Cmdr
- Fast!

and more

my duplicated ticket #55861

**Comment by songlairui** [comment]
no terminal was better than vscode-terminal

**Comment by LujunWeng** [comment]
Hi guys,

I am hacking vscode to make the integrated terminal a standalone app. Check my preliminary work out: https://github.com/LujunWeng/vscterm. Also, there is an installer for win32-x64: https://github.com/LujunWeng/vscterm/releases. 

What I have done:
1. Removed other UI components and only the terminal left. Figuring out ways to remove those UI components' services.
2. I also removed builtin extensions to reduce the whole volume of the repo.


Cheers...

**Comment by paralin** [comment]
@LujunWeng 34,000 commits... did you fork vscode itself? would suggest making a lighter repo for it starting out.

**Comment by LujunWeng** [comment]
@paralin yes. Basically, I want to keep the history, but it must take long time to clone the whole repo. Yeah. Thank you. It should be a lighter repo.

**Comment by paralin** [comment]
@LujunWeng You have stripped down the repo a lot, already yeah? If you don't care about history, you can wipe it with a rebase. 

Another approach might be to develop as you have for your example app (the extracted terminal) and slowly migrate the terminal code out into another library.

**Comment by LujunWeng** [comment]
@paralin Thank you for suggestion. I did rebase.

**Comment by nojvek** [comment]
VScode Terminal as a stand-alone app would be great to have but I can see the architectural difficulties it may present and vscode team may be hesitant to it. 

Being able to undock the terminal panel as it’s own maximizable window and switch from dropdown to tabs should allievate a lot of pain. 

In multiple monitor mode, it’s nice to have terminal separately on it’s own screen.

**Comment by Tyriar** [comment]
@nojvek that's definitely on the list of things I want to do in the near future.

**Comment by sagebind** [comment]
> Being able to undock the terminal panel as it’s own maximizable window and switch from dropdown to tabs should allievate a lot of pain.

Agreed. In particular I think that solving #10546 would minimally solve a lot of asks simultaneously in a generic way. For example, moving the terminal to a separate window could be just a second regular Code window with a terminal tab in it.

**Comment by SokHeangKhun** [comment]
I have 2 monitors and I want to dock out the terminal, output and etc which dock on the bottom of the editor to be in my another monitor with its own window which give me clearly space to work. When do I get this feature in Visual Studio Studio Code?

**Comment by wrightleft** [comment]
> I have 2 monitors and I want to dock out the terminal, output and etc which dock on the bottom of the editor to be in my another monitor with its own window which give me clearly space to work. When do I get this feature in Visual Studio Studio Code?

Exactly what I'm looking for!

**Comment by mikemaccana** [comment]
> When do I get this feature in Visual Studio Studio Code?

When you send the pull request? 😉

**Comment by MatthewCash** [comment]
This feature would be amazing. The only terminal I can stand to use is Hyper, and I feel that everyday something about it goes wrong, the text disappears, random errors, or weird margins. I will sometimes open a VScode terminal fullscreen just to use it alone.

**Comment by mikemaccana** [comment]
@matt`hewcash I like Hyper and vscode too, but if you want a *stable* feature filled Windows shell try Terminus. It works (including important things like readline), has iterm themes, defaults to Powershell core, and all the customisability of Hyper.

**Comment by Tyriar** [comment]
@mikemaccana FYI Hyper, vscode and Terminus all use the backing technology that I help maintain for vscode (xterm.js and node-pty), so the core terminal experience should be pretty similar across each.

> When you send the pull request? 😉

Unfortunately this issue is far too large to get externally so I recommend against anyone trying to make a PR.

**Comment by Luummii** [comment]
1. Create new folder in your project(for example folder name - terminal)
2. Open folder "terminal" in new window
3. Open new terminal there
4. Run command cd ../
5. Resize terminal on full window
6. Close terminal in your project
7. Work! 

Is simple and cheap. :)

**Comment by mikemaccana** [comment]
@Tyriar most of the bugs at a higher level than node-pty - I run practical tests of most of the Windows terminals each month with links to any big issues that stop powershell working. https://github.com/mikemaccana/powershell-profile#prerequisities-for-any-nix-user-who-wants-to-use-powershell

**Comment by nojvek** [comment]
@Tyriar,

In your opinion, the terminal panel to be toggle-able as a separate
electron window/main workbench is also far too reached to be a PR?

I feel that is an incremental baby step delivers 80% of value with 20%
effort.

**Comment by Tyriar** [comment]
@nojvek yes that will be very involved. Our top 👍 voted issue is actually "floating window" support which is essentially that and is on the roadmap so someone will probably investigate this within the next 12 months https://github.com/Microsoft/vscode/wiki/Roadmap#workbench

**Comment by lucaswunder** [comment]
Is not it better to open another vscode window with the file you want or terminal cmd?

**Comment by Pomax** [comment]
I'm not sure why this keeps getting turned into "this is part of turning panels into floating panels". The terminal is special enough that it _doesn't_ need to wait for panels in general to made floating because unlike every other panel, the terminal _does not communicate with anything else_.

Unlike every other panel, what users do inside the terminal has literally no bearing on VS Code outside of what "running a terminal outside of VS Code" does (e.g. files popping into existence or disappearing). While the work to make panels in general is a heck of a lot of crazy re-architecting, making _just the terminal_ something we can pop out is _way_ easier because there is no data exchange that needs to happen in terms of what's in the terminal.

This is something that someone with a week of time and enthusiasm could probably do (with guidance, of course), and is extremely worth _not_ treating as "part of the floating panels" work. It expressly is not. It's way easier.

**Comment by Tyriar** [comment]
@Pomax

> it doesn't need to wait for panels in general to made floating because unlike every other panel, the terminal does not need to tie into, or communicate with, anything else (unlike problems, or output, or debug).

This is not true, it touches a lot of components and a lot of components touch it.

> This is something that someone with a week of time and enthusiasm could probably do (with guidance, of course)

It could certainly be hacked together by someone with enthusiasm, but it's not as simple of a change as you might think to _do right_. AFAIK the plan is to use the terminal as the starting place for pulling a part of a VS Code window into another window.

I bet I'm more keen to see this done than you are, but we need to wait until it gets on the plan. The best thing you can do it :+1: the issue as that's one of the indicators that feeds into our prioritization process.

**Comment by musm** [comment]
I see this issue basically also just being able to send code to an external terminal in a more robust fashion.

Could we note take advantage of the new  https://github.com/microsoft/terminal to make the "sending code to an external terminal" more functional?

**Comment by SHxKM** [comment]
Since issues that ask for a "detach terminal" feature are getting closed in favour of this (even though it looks like a separate request), I'd like to know that that's something I would very much like to have.

**Comment by Tyriar** [comment]
@SHxKM for detaching see https://github.com/microsoft/vscode/issues/10121 and https://github.com/microsoft/vscode/issues/10546

**Comment by hgouveia** [comment]
Detach is different request, i actually want an standalone app of the vscode terminal, this is terminal is one of the best out there tbh,  with nice feature like customization, shell selection, split, tabs

and i would like to be able to use outside vscode

![image](https://user-images.githubusercontent.com/668866/60656942-b7384000-9e50-11e9-83d1-262c785ddd0f.png)

**Comment by Tyriar** [comment]
@hgouveia what you're describing is what I consider this issue to be.

**Comment by donaldpipowitch** [comment]
> i actually want an standalone app of the vscode terminal,

Me, too. Would also be nice if it could have its completely own extension ecosystem and maybe create a foundation for sharing certain things (e.g. extension handling) so other apps could use that as well. (There seems to be at least ["some" interest](https://twitter.com/Martin_Adams/status/1136535126723256322) in this.)

**Comment by Tyriar** [comment]
The way I see this issue playing out is not having it completely standalone, but allow creating a terminal through CLI args `code --term` to launch just a terminal not associated with any workspace.

> Would also be nice if it could have its completely own extension ecosystem

xterm.js the underlying component for the terminal is just finalizing it's "addon" system in v4, that is the extensions for the terminal itself such that you can build on top of it without putting everything into the core https://github.com/xtermjs/xterm.js/releases/tag/3.14.0. Being able to plug in xterm.js addons through VS Code is not really a direction that I'd want it to go because xterm.js addons run on the renderer, not the extension host (which protects the renderer process/DOM from naughty extensions). They do let you write functionality that could be used in multiple terminals though.

As for VS Code's "base" as a platform, there was a discussion about this very early on and everyone was pretty against it as it means we would lose agility due to having low-level APIs we can't break. Things like new tree view that boosts perf would be much more difficult in such a world.

**Comment by nojvek** [comment]
`code --term` would be super nice. I do wonder how much RAM it will end up using. If I have both `code` and `code --term` open at the same time in different windows. Does that make two instances of electron with at-least 150MB of RAM/window ?

iTerm uses 164MB for the base and about ~5MB per tab so may be it's comparable. 

I know VSCode and especially you care a great deal about memory/cpu/UI lag a lot, unlike slack which ends up using multi gigabytes of RAM. 

So excited for `code --term` when it ships.

**Comment by Tyriar** [comment]
The memory usage would likely be much smaller than 150mb as we could avoid creating most of the services. I think the memory per tab would be around 5-15mb, but it depends on what your scrollback is set to as we init arrays for the full buffer ahead of time.

**Comment by aleksijohansson** [comment]
Looking forward to this!

**Comment by lolimay** [comment]
Any updates?

**Comment by wondernate** [comment]
Would love this!!!  Currently I am using a second instance of VSCode to run my terminal but I loose the ability to link to my current code in the errors.

**Comment by datat** [comment]
It would be great to see the arrow keys working in term much like RStudio and PowerShell ISE.

**Comment by Hustcw** [comment]
Looking forward to this. VScode terminal is the best one I used in windows.

**Comment by gregzanch** [comment]
Would love this.

**Comment by badelectron77** [comment]
This would be great!

**Comment by misarfraz** [comment]
This would be awesome!

**Comment by Giabanga** [comment]
Popping out the terminal to a separate window (and even editors etc) for dual screen support is a must have imho. For me, it is the main drawback compared to Visual Studio and other IDEs

**Comment by thiefuniverse** [comment]
I really want to use terminal in vscode by emacs.... 😃  Terminal in vscode is best terminal I see which is in one editor.

**Comment by Ayanmullick** [comment]
> I see this issue basically also just being able to send code to an external terminal in a more robust fashion.
> 
> Could we note take advantage of the new https://github.com/microsoft/terminal to make the "sending code to an external terminal" more functional?

One can configure an 'external terminal' in VSC 'Settings' and open one with Ctrl+Shift+C in Windows. It'd be great if one could point the external terminal to the new Microsoft Terminal, make that the active terminal in VSC  and run code from the VSC script pane[F5,F8 etc.] in the Microsoft terminal. Currently, one cannot run **PowerShell** script from the VSC script pane in the external terminal but I hear it's doable in Angular etc.

**Comment by ultraGentle** [comment]
In the interim, before a true extracted terminal, would it perhaps be possible to add an option to **configure VSCode as the "external" terminal?**   

Sounds contradictory, but it would work like this: **when you open an external terminal, it's actually a new instance of VSCode with the terminal maximized.**  It could even have all non-terminal extensions off, for the sake of efficiency.

**Comment by Tyriar** [comment]
@ultraGentle that might be something when can do when we allow the terminal to be pulled out of the main window, chances are it wouldn't use the external terminal commands to do so though. https://github.com/microsoft/vscode/issues/10121

**Comment by ultraGentle** [comment]
@Tyriar thanks for the response.  Just to clarify, I meant that since some folks are already creating a new instance of VSCode just to use the terminal in a separate window, there could be a shortcut for that behavior to mock the effect of an extracted terminal.  

It's not a big deal, since that's already easy to do with a few commands in a row (Cmd+N  --> Ctrl+` --> Toggle Maximize Panel).  Just an idea to fold that process into a single command. 

Anyway, no worries, and cheers!

**Comment by codepunkt** [comment]
When you use development containers, but would love to have your vscode terminal on a different screen, you're all out of luck.

You'd either need a "detach" option for the terminal of the current vscode instance or close the terminal and docker exec yourself into the development container from hyper.

**Comment by tigerinus** [comment]
Would love to see VSCode integrated terminal detachable and/or able to start it as a standalone app.

Windows Terminal? Before it has some degree of integration with VSCode, I will say: Thanks. But no, thanks. BTW, WT doesn't support bold font which I love about VSCode Terminal.

**Comment by darkguy2008** [comment]
I know most of you are hyped about the terminal but Windows is GUI. There's plenty of terminal apps out there if one wants to use it. The main objective of this issue, as I take it and feel it and the reason why I stumbled upon this issue, too, is that it's really tiresome to be resizing the terminal window (or hitting a hotkey) **everytime** I make a build and need to see a full-screen terminal to view debug output as my current app requires the terminal to display stuff and can't use the debug console (don't ask, it's just like that).

So, the main concern _should be_ detaching the terminal from the VSCode window so I can at least open it in a secondary monitor (I've got 4, go figure) to view terminal output with more real estate to avoid resizing the panel back & forth. I have to expand the terminal to view debug, then resize back to edit, then expand to view my changes, then resize to edit... man, it's horrible.

Windows Terminal, HyperTerm, iTerm, whatever, are all cool, but I think they're outside of the scope of the real issue here.

**Comment by VandalPaulius** [comment]
I second that. It would be especially useful when having multiple monitors with different sizes and debugging anything on VSCode.

For the quick and dirty way, I wonder if it's possible to write a plugin that just mirrors output from terminal to the other terminal process

**Comment by pradhyu** [comment]
> I really want to use terminal in vscode by emacs.... 😃 Terminal in vscode is best terminal I see which is in one editor.

open a terminal 
emacs -nw 

I had to fix lots of keybindings to make sure the keybindings in vscode doesn't interfere with emacs. 
Just try to use emacs key bindings if anything doesn't work figure out what keybinding is interfering then change the (where binding) condition for that keychord to add && !terminalFocus.

**Comment by lloeki** [comment]
> The way I see this issue playing out is not having it completely standalone, but allow creating a terminal through CLI args code --term to launch just a terminal not associated with any workspace.

Sounds good, except maybe for one UX point: if I'm not mistaken, any update of VS Code would kill the terminals as well. But I'm all for it if it makes the ball move forward.

**Comment by lloeki** [comment]
> So, the main concern should be detaching the terminal from the VSCode window so I can at least open it in a secondary monitor (I've got 4, go figure) to view terminal output with more real estate to avoid resizing the panel back & forth

This use case is definitely part of the discussion (and possibly part of the plan IIUC), but such a terminal is bound to a workspace. This issue right here *also* covers the use case of having a terminal untied to any workspace (and maybe as a separate program). This workspace-less state kind of exists when you start a new window (which starts a blank, empty workspace which can have terminals started in that state).

> I know most of you are hyped about the terminal but Windows is GUI

So is macOS, having a decent terminal is a significant boost. Having consistent, well-behaved terminals throughout the OS is a definite boost day to day.

>  I have to expand the terminal to view debug, then resize back to edit, then expand to view my changes, then resize to edit... man, it's horrible.

My current workaround is to use ``^` `` or `⌘J` to quickly toggle the bottom panel back and forth (after having resized it to a good enough size) Quake-console-style.

**Comment by patrislav1** [comment]
> Would love this!!! Currently I am using a second instance of VSCode to run my terminal but I loose the ability to link to my current code in the errors.

Relating errors / warning messages to code currently worked on in the editor is also the same use case for me. Right now I work around the terminal limitation by spanning one VS Code window over two monitors, select "Move Panel Right" from the terminal's context menu, and move the panel boundary such that the editor/source tree window is on the left screen, and the terminal on the right screen. That's a ugly workaround that requires both monitors having the same resolution, messes up the window title split between screens, and is not possible in full screen mode - but it's way better than having just the small terminal panel at the bottom.

**Comment by WTassoux** [comment]
This would be amazing, especially with the latest terminal improvements. VSCode terminal feels snappier than PuTTY !

**Comment by micarex** [comment]
Separate terminal on second screen would definitely be useful.

**Comment by stephancasas** [comment]
+1

**Comment by qhyun2** [comment]
Here is another ugly work around using Tmux:
https://qhyun.medium.com/functional-external-vscode-terminal-using-tmux-485962384f56

**Comment by Bjorn80** [comment]
Would love to see this happen. Being able to maximize the code on one screen while viewing the results (Terminal) on my second screen. 

Make it so.

It is the way.

**Comment by lrnv** [comment]
While readding the comments here, i notice that there are two feature requests that are different but still lives on the same issue: 

- Begin able to use the Vscode terminal standalone
- Alowing the vscode terminal, still attached to the vscode session, to be poped-out as an independant *window*, mainly to put it on another screen to fully use screen real estate. 

These two features are clearly different. Since I only want the second one, which looks easier to implement (but might not be, i'm not aware of vscode internals), maybe we should split the issue in two ? 

Furthermore, feature requests asking for only one of the two are systematically closed as duplicates of this one.

I think we should split.

**Comment by yume-chan** [comment]
@lrnv I really don't want to repeat once more (https://github.com/microsoft/vscode/issues/34442#issuecomment-508214479, https://github.com/microsoft/vscode/issues/34442#issuecomment-403465154) and send another spam email to all subscribers, but...

For detaching panels/views to separate windows we already have #10121, but it's very difficult, if not impossible, to implement.

So basically, the idea now is extracting the terminal part to a standalone executable, or allow an instance of Code to run in "terminal mode" (https://github.com/microsoft/vscode/issues/34442#issuecomment-508834722) so we can have both standalone terminal app AND detachable terminal at once.

**Comment by Tyriar** [comment]
@lrnv yes I'm aware of the confusion and while I have been tracking "Begin able to use the Vscode terminal standalone", I know that the upvote count is definitely misleading.

As @yume-chan calls out, https://github.com/microsoft/vscode/issues/10121 is the item for being able to pull the terminal _view_ out. There's another piece of this in https://github.com/microsoft/vscode/issues/10546 which would hopefully allow terminal _tabs_ to be pulled out as well. I might update the OP's comment since GitHub doesn't support pinned comments (in issues anyway).

**Comment by lrnv** [comment]
@Tyriar Thanks a lot for the clarification in the OP.

**Comment by olegKusov** [comment]
4 years have passed and the feature has not been implemented

**Comment by jacky-liang** [comment]
please implement this!

**Comment by ridhwaans** [comment]
If Microsoft Terminal is installed, it is also equally useful to add open this current tab or window in MSFT Terminal option in the right click context menu in VS Code

**Comment by Tyriar** [comment]
@ridhwaans I created a small extension to give some context menu options for Windows Terminal recently: https://marketplace.visualstudio.com/items?itemName=Tyriar.windows-terminal

**Comment by Ayanmullick** [comment]
> @ridhwaans I created a small extension to give some context menu options for Windows Terminal recently: https://marketplace.visualstudio.com/items?itemName=Tyriar.windows-terminal

Could one run a code selection from the VSC script pane on the Windows Terminal using F8?

**Comment by Tyriar** [comment]
@Ayanmullick the extension doesn't do it right now, but if you can tell WT to run a command (by launching a new tab?) then it should be possible. The might be complications around getting it to work properly with the format of the command though (eg. the pwsh command looks different to the bash command).

**Comment by musm** [comment]
It seems like a better solution to the "extract integrated terminal" problem is to instead utilize the Windows terminal and create a communication protocol to send data to and from. As far as my needs are concerned this would solve what I really want from the core of this issue.

**Comment by patrislav1** [comment]
This is my workaround (for gcc compiler messages): https://github.com/patrislav1/vmake

**Comment by Ayanmullick** [comment]
> @Ayanmullick the extension doesn't do it right now, but if you can tell WT to run a command (by launching a new tab?) then it should be possible. The might be complications around getting it to work properly with the format of the command though (eg. the pwsh command looks different to the bash command).

WT runs the user's PowerShell profile when launching a new PowerShell tab. What should one add in the profile to enable that communication between the VSC script pane and the WT?

**Comment by cassiozen** [comment]
It would be awesome if running "code ." would just augment the current terminal session with the rest of VSCode.

**Comment by TylerLeonhardt** [comment]
I just wanna add that I would love for "quake mode" to be a part of this :) 

It seems folks liked the idea over in the Windows Terminal space: https://github.com/microsoft/terminal/issues/653

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
The integrated terminal should merely be [Windows Terminal](http://github.com/microsoft/terminal) hosted as a toolbox window.

**Comment by Tyriar** [comment]
@BEEDELLROKEJULIANLOCKHART this has been suggested before but it's not possible as it wouldn't work in many of the places where VS Code runs: Linux, macOS, Codespaces (Browser, Android, iPad, etc.). Besides it would be quite the undertaking to do that adaption just for Windows and we would actually regress in a lot of areas. That's without considering the potential downsides of hosting such a toolbox window inside a toolbox app which I'm sure there are plenty.

**Comment by borekb** [comment]
> The integrated terminal should merely be Windows Terminal hosted as a toolbox window.

I think the opposite – that Windows Terminal should be just wrapped VSCode's terminal 😄.

**Comment by sagebind** [comment]
The issue in question here is making it possible to run the vscode terminal in a standalone window. What's that got to do with Windows Terminal? I doubt Windows Terminal wants to ship all of vscode just for a terminal, and conversely vscode doesn't need Windows Terminal because the integrated terminal is way more integrated into the vscode experience.

They should just stay separate things and people can use what they prefer. A little competition never hurt any application anyway.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
http://github.com/microsoft/vscode/issues/34442#issuecomment-901921248, I have never observed compettition within one legal entity to be benficial for products that must be paid for. http://web.telegram.org/z/ and http://web.telegram.org/k/ are rather good examples of this because much has been wasted by designing two almost-identical websites, and thus 2 code-bases.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
https://github.com/microsoft/vscode/issues/34442#issuecomment-901850776, indeed, naively I did not consider whether [Windows Terminal](http://github.com/microsoft/terminal) is supported by [Microsoft](http://github.com/microsoft) for operating-systems other than Windows, irrespective of its obviously-denotive name.

If theoretical utilisation of [Windows Terminal](http://github.com/microsoft/terminal) would be functionally regressive, surely as what was stated at http://github.com/microsoft/vscode/issues/34442#issuecomment-901916458 would be better for [Microsoft](http://github.com/microsoft).

**Comment by hgouveia** [comment]
Years ago i was supporting this ticket and asking for this feature, but since [Windows Terminal ](https://github.com/microsoft/terminal) was released, i am not longer feel the need to this feature, since is a fantastic substitute and does all i want, i would vote to close this , and instead use this time to improve more [Windows Terminal ](https://github.com/microsoft/terminal) making it faster and better and better

**Comment by tigerinus** [comment]
I also vote to close this. Windows Terminal is good enough and it works with VSCode fine. 

I'd vote for the `Quake Mode` but that feels like a different feature request to be filed.

**Comment by lrnv** [comment]
I agree, but please follow @Tyriar precision and do NOT close #10121 as duplicate of this in the future... #10121 needs to stay open, even if we cannot talk in it anymore.

**Comment by Tyriar** [comment]
There are plenty of people that still want this so I'm not closing it, but I think things have changed a little since WT came out as the need for a solid conpty-backed terminal on Windows is now covered. Here's what I feel this issue now encapsulates:

- Being able to have a lightweight VS Code window with a fullscreen terminal in it, lightweight meaning minimal memory. This is more than https://github.com/microsoft/vscode/issues/10121 which would just be allowing dragging the panel of an existing window out.
- These lightweight terminals could be launched with something like `code --term` or via a command.
- The window could be "upgraded" by running `code -r <folder>` which would open a folder in that window and reconnect the terminals that were previously there.
- Potentially bringing quake mode in which would use one of these lightweight terminals.

**Comment by darkguy2008** [comment]
I disagree with the decision of closing this issue and also thinking that Windows Terminal is a drop-in replacement for this feature. Windows Terminal may be good and all, but it is NOT what we're looking for (at least the ones interested in seeing this issue come to light):

As for @Tyriar 's points, the first three do encapsulate what we want but the last one IMO is just fancyness (and personally I've always disliked those quake-like terminals).

Windows Terminal is a good app but it feels heavy and bloated as heck, unlike VSCode. Considering there are other similar (but not as lightweight or preferred as VSCode's) like Tabby (which is Electron-based) one does wonder why is it so hard for the VSCode team to implement a feature with a small click menu that says "Detach terminal" and just opens VSCode in "terminal mode"? The ones that are asking for this feature only want a good terminal (like the VSCode team has achieved with it so far) that's lightweight and feels familiar.

Therefore, I disagree completely on closing this issue or duplicates and marking this as a wontfix, because it feels as if MS doesn't want to listen to its community and is what actually creates more forks, more disappointment, more fragmentation and more unhappy users along the way.

**Comment by Tyriar** [comment]
> one does wonder why is it so hard for the VSCode team to implement a feature with a small click menu that says "Detach terminal" and just opens VSCode in "terminal mode"? 

@darkguy2008 the slowness is a combination of resources, other priorities (I have almost 500 issues assigned atm 😅) and other things that it's blocked on. For example #10121 poses architectural problems since VS Code's terminal integrates with so many internal services, the naive approach would be to initialize all these services but that would mean a lot of extra memory and cpu which doesn't really meet the quality bar we're aiming for.

We have been smashing through some major features in the terminal recently though (tabs, profiles, terminals in editor area, process reconnect/detach), so this doesn't seem so far off anymore 🤞

**Comment by tigerinus** [comment]
A VSCode with only integrated terminal sounds more bloated than Windows Terminal or any terminal on other platforms (rxvt, konsole, Mac term, etc.) even it is more lightweight than a VSCode with all extensions enabled.

Behind the scene VSCode is built using electron.js, with a Chromeium based rendering process and a number of nodejs based backend processes. Even the `code main` can consume at least ~90MB:

![image](https://user-images.githubusercontent.com/7172560/130089707-b80b1f93-a7d9-4f70-a75c-8119ce9b1272.png)

**Comment by darkguy2008** [comment]
> A VSCode with only integrated terminal sounds more bloated than Windows Terminal or any terminal on other platforms (rxvt, konsole, Mac term, etc.) even it is more lightweight than a VSCode with all extensions enabled.
> 
> Behind the scene VSCode is built using electron.js, with a Chromeium based rendering process and a number of nodejs based backend processes. Even the `code main` can consume at least ~90MB:
> 
> ![image](https://user-images.githubusercontent.com/7172560/130089707-b80b1f93-a7d9-4f70-a75c-8119ce9b1272.png)

Well, another point for me: https://github.com/microsoft/terminal/issues/3843 it seems that even in the same Windows platform, Windows Terminal isn't compatible. It's too bad that to download it requires the Microsoft Store (so this excludes a lot of users that don't want or don't have the store - unlike VSCode) and also it's uncompatible with Windows LTSC, which is the enterprise version that some companies do use. For instance, my work laptop uses LTSC (and I love it too, considering it isn't bloated as heck) and I'm unable to use the WT for this same reason, just like other users who would be excluded due to this reason, not to mention the lack of cross-platform of Windows Terminal.

It's a good app, but again, not a drop-in replacement.

I'm not very concerned about the memory usage (RAM is pretty cheap nowadays) but more about the UX, the speed/snappyness of it that I've yet to find in any other standalone Electron-based terminal emulator.

Plus, with the session persistence feature that currently VSCode has in place, I had expected that detaching the session, creating a new window and reloading the session there with just the terminal wasn't too much of a hassle now.

Thankfully, there's people like @Tyriar who actually has a point and now sees this feature as a possible one in a near future (hopefully), so thanks for taking us into account!

**Comment by Tyriar** [comment]
@tigerinus right, I would expect the eventual memory usage of a terminal window to be similar to the process explorer window which is sitting at 82mb in your screenshot, plus the memory xterm.js takes for its buffer. What we would not want it something closer to a normal window like the 163mb one in your screenshot.

Clearly WT is capable of being faster since it's closer to the metal, but beyond the overhead of Electron, the [xterm.js](https://github.com/xtermjs/xterm.js) is a very good citizen wrt memory and CPU.

---

I don't think we need to get into a philosophical debate on "lightweightedness" and whether WT or VS Code is better. A lot of people get notifications for this issue and that doesn't feel productive.

**Comment by sagebind** [comment]
> Windows Terminal is a good app but it feels heavy and bloated as heck

I'm not sure what gives you that impression. Windows Terminal will probably forever be lighter and more efficient than VSCode's terminal, because the former is written in C++ without any extra layers on top, while VSCode depends on Electron.

> one does wonder why is it so hard (and has taken way too long... 4 years, seriously?) for the VSCode team to implement a feature with a small click menu that says "Detach terminal" and just opens VSCode in "terminal mode"?

Software doesn't work like that. It's easy to add a menu that says "Detach terminal", but it is hard to make clicking that menu item _actually do what you ask_. Like @Tyriar mentioned, having a detached terminal window goes against the entire VSCode architecture as it stands and likely requires significant work to be able to disentangle the integrated terminal codebase from the context of a VSCode workspace window.

And speaking from experience as a maintainer of my own software, complaining about timelines is unlikely to be a motivator to deliver the feature you want any quicker.

**Comment by darkguy2008** [comment]
Well, I still think this issue should not be closed and the feature request is pretty valid and has some good points (i.e. cross-platform, IDE integration) plus other features that WT can't serve because it basically falls out of scope and it excludes a lot of users that VS is actually taking.

Again, I'm not concerned about RAM usage... If an user can run VSCode, it's more than safe to assume that it has plenty of RAM to support an external, detached terminal if it wants to (also, knowingly it'll take RAM). I don't really see why this is a problem here.

**Comment by tigerinus** [comment]
@Tyriar - I've opened a different feature request at https://github.com/microsoft/vscode/issues/131188 regarding launching external terminal and attach a VSCode terminal session to it.

Thanks for all the hard work!

**Comment by ErnestoBorio** [comment]
This is just a workaround I'm using and it's only for unix-like systems and just for the terminal panel, but you can open a new terminal outside vscode, and get its device path with `who am i`.
Suppose it's `/dev/ttys001`.
Then in the vscode terminal you run your process, let's say `npm run start:dev &> /dev/ttys001`, then you get the terminal output in a separate window.
You could just run the task in that terminal directly, but for tasks that attach to the debugger, this can only be done from vscode integrated terminal.
Hope this helps someone.

**Comment by blueray453** [comment]
When we run 

`$ code .`

then <kbd>Ctrl+Shift+P</kbd> > `Terminal: Create New Terminal in Editor Area`, it looks like:

![Screenshot-2021-10-01-11:47:48](https://user-images.githubusercontent.com/855297/135571812-58f3cfd7-25fc-4583-9c29-909bf1b2c4d6.png)

Great Work!

Now if we can run `code .` and then `Terminal: Create New Terminal in Editor Area` in one go, then we **sort of** got a terminal.

**Comment by pasqualirb** [comment]
> I think the opposite – that Windows Terminal should be just wrapped VSCode's terminal 😄.

Quoting what @borekb said, I think this is actually something to be considered. I don't think Windows Terminal should be abandoned, though.

Regarding performance, it's reasonable to say that a good performance can be achieved by using asynchronous operations with rendering optimizations. I'm actually skeptical that Windows Terminal would be too superior in user-perceived responsiveness when compared to such a program. A browser engine like V8 can be viewed as an optimization engine that _when used right_ can perform very well.

The main advantage of this approach is having an **official cross-platform terminal** that is highly customizable and integrates well with vscode.

**Comment by Tyriar** [comment]
> I think the opposite – that Windows Terminal should be just wrapped VSCode's terminal 😄.

I actually suggested this to the WT team when it was in the early stages but that ideal ultimately fell apart since the team wasn't that familiar with web tech and they didn't believe the sort of performance/overhead they would need to get it shipped within Windows (which happened with W11) would be possible.

**Comment by fabiospampinato** [comment]
> and they didn't believe the sort of performance/overhead they would need to get it shipped within Windows (which happened with W11) would be possible.

I shouldn't say this, but ironically it's most probably possible to make a web terminal that's 10x faster than WT.

**Comment by alvipeo** [comment]
Don't laugh, but I find regular cmd.exe on Windows my best terminal (for running npm dev processed like npm run start, etc). But I'd **really** to have it on my second monitor when working in codespaces.

**Comment by bradisbell** [comment]
@alvipeo Absolutely.  I do the same thing.  And for me, invariably the regular command prompt is faster anyway.  I often have to wait for characters to appear on the VS Code terminal, depending on what's going on at that moment.  Pretty wild on a beefy machine this day and age.

**Comment by alvipeo** [comment]
Just to be clear - it's not just merely performance of it for me, it the *second* monitor. I can have the whole screen with terminal output, not that small area at the bottom of VS Code.

**Comment by xcodebuild** [comment]
CodeTerminal here, now we can use Terminal from vscode as a standalone app!

https://github.com/xcodebuild/CodeTerminal

![image](https://user-images.githubusercontent.com/5436704/162680330-75a0fccd-012b-41df-bb58-cc073881be53.png)

🚀 Feel free to try it out.

**Comment by darkguy2008** [comment]
> CodeTerminal here, now we can use Terminal from vscode as a standalone app!
> 
> https://github.com/xcodebuild/CodeTerminal
> 
> 🚀 Feel free to try it out.

Okay so while this isn't exactly the original request (_detaching_ the existing terminal window into a separate one to put it on another monitor), this is **AWESOME!** The VSCode terminal in itself, in its own standalone form, is pretty good and solid. This looks like an amazing project to follow, thanks for sharing! I just gave it a try and it worked pretty well! :)

**Comment by notblank00** [comment]
It's a bit too slow/laggy on Linux but I'd love to see this too

**Comment by Emahhh** [comment]
I'd love to be able to drag the vscode terminal out in a new window, so that i can take advantage of my second monitor. It doesn't need to be a standalone app tho.

**Comment by iFwu** [comment]
It become more necessary since history of VSCode integrated terminal use different file from other terminal apps.


> @Tyriar should it be using the same history file?
_Originally posted by @meganrogge in https://github.com/microsoft/vscode/issues/145296#issuecomment-1112671691_

**Comment by Tyriar** [comment]
@iFwu reopened that issue to get clarity there, it should be using the same file.

**Comment by felipecrs** [comment]
Well, you can do something like this:

https://user-images.githubusercontent.com/29582865/178532158-e4131237-e7c3-449f-b772-9281c88fb072.mp4

It isn't exactly seamless though.

**Comment by LightningStark** [comment]
> I'd love to be able to drag the vscode terminal out in a new window, so that i can take advantage of my second monitor. It doesn't need to be a standalone app tho.

you can use [Schnell Console](https://github.com/thevoyagingstar/Schnell) alongside!

**Comment by ErikHumphrey** [comment]
> CodeTerminal here, now we can use Terminal from vscode as a standalone app!
> 
> https://github.com/xcodebuild/CodeTerminal
> 
> ![image](https://user-images.githubusercontent.com/5436704/162680330-75a0fccd-012b-41df-bb58-cc073881be53.png)
> 
> 🚀 Feel free to try it out.

Darn, that just got archived

**Comment by salahusama** [comment]
One thing people can do to get the VS Code terminal as a "standalone" terminal is as follows:
1. Open a new VS Code window
2. Open the terminal
3. Move Terminal into Editor Area
4. View > Appearance > Zen Mode
5. View > Appearance > Centered Layout _(uncheck)_

**Note 1:** Zen Mode makes the window full screen, but if you exit full screen, you should still get the right view.
**Note 2:** I only tested this on macOS, but I imagine it would be a similar behavior for Windows and Linux.

<img width="1792" alt="Standalone VS Code Terminal Fullscreen" src="https://user-images.githubusercontent.com/12504419/191066777-302c3557-a329-48e6-afdc-440058b9255e.png">

<img width="1253" alt="Standalone VS Code Terminal Not Fullscreen" src="https://user-images.githubusercontent.com/12504419/191067303-bc9571f7-6b17-4d8e-af95-15387865c34f.png">

**Comment by TrentDevelopsMS** [comment]
> One thing people can do to get the VS Code terminal as a "standalone" terminal is as follows:
> 
> 1. Open a new VS Code window
> 2. Open the terminal
> 3. Move Terminal into Editor Area
> 4. View > Appearance > Zen Mode
> 5. View > Appearance > Centered Layout _(uncheck)_
> 
> **Note 1:** Zen Mode makes the window full screen, but if you exit full screen, you should still get the right view. **Note 2:** I only tested this on macOS, but I imagine it would be a similar behavior for Windows and Linux.
> 
> <img alt="Standalone VS Code Terminal Fullscreen" width="1792" src="https://user-images.githubusercontent.com/12504419/191066777-302c3557-a329-48e6-afdc-440058b9255e.png">
> 
> <img alt="Standalone VS Code Terminal Not Fullscreen" width="1253" src="https://user-images.githubusercontent.com/12504419/191067303-bc9571f7-6b17-4d8e-af95-15387865c34f.png">

I feel like this is the answer to getting to a "Pop-out" terminal. It's essentially VSCode with all the other features hidden. Probably not as optimal performance-wise but are we splitting hairs here?

On Mac I prefer to have my terminal on the desktop layout (I know, like a fiend) so I customized the layout and disabled visibility of the "Activity Bar" and "Status Bar".

**Comment by cameronroe** [comment]
Would also like a simple button on vscode to pop out the terminal into a new window. Cheers.
![image](https://user-images.githubusercontent.com/1614348/206844005-2c9ddfb0-15cc-404a-91da-9a90ebd65624.png)

**Comment by Tyriar** [comment]
@TrentDevelopsMS we wouldn't want to ship a sub-par version that needs to load the entire workbench as we'll inevitably get complaints about memory usage. Plus, you can essentially do that already by creating a new workspace and maximizing the panel, it's just a little less convenient.

**Comment by PhilParisot** [comment]
Any news? Is this issue on any roadmap? That's how much I love the VS Code terminal, and I think I'm not alone.

**Comment by Tyriar** [comment]
@philparisot [this is the official roadmap](https://github.com/microsoft/vscode-wiki/blob/main/Roadmap.md) which will probably get an update soon. It's not there yet but I expect it will be within the next 1-2 years. Intellisense in the terminal (https://github.com/microsoft/vscode/pull/171648) and leveraging shell integration smarts in other ways is current big focus for the terminal right now.

**Comment by felipecrs** [comment]
By the way, I think something like a new CLI flag would be more than enough:

```
code --only-terminal
```

Which opens the VS Code window with the terminal maximized, and panels collapsed.

**Comment by rokejulianlockhart** [comment]
@felipecrs, I'd make it `code --terminal` for simplicity.

**Comment by jasondamour** [comment]
I think the main draw of extracting the integrated terminal (over using a standalone terminal application, even on which shares a codebase with VSCode) is the ability to preserve long-running operations when switching workspaces. If I start a script in one workspace then want to switch workspaces, it closes that terminal. I'd like to be able to pop-out the attached terminal and change workspaces without stopping that shell.

**Comment by Tyriar** [comment]
@jasondamour FYI you can run the detach session command and attach session in the other window, or just drag it over before closing the old window https://code.visualstudio.com/docs/terminal/advanced#_moving-terminals-between-windows

**Comment by starball5** [comment]
Related on Stack Overflow: [Is it possible to open integrated terminals from VS Code in a separate window?](https://stackoverflow.com/a/76214341/11107541)

**Comment by Chrislann1974** [comment]
Scan phone number

**Comment by Chrislann1974** [comment]
Scan

**Comment by risserlabs-com** [comment]
This would be amazing. I sometimes want to run a terminal without running vscode. I really like the vscode terminal.

**Comment by zocky** [comment]
Request to close this issue. It mixes up two completely separate concerns: 
(a) using vscode's integrated terminal as a replacement for the poor native terminal implementations in Windows.
(b) detaching panels so they can be placed e.g. on a secondary monitor

Out of these, (a) is really not vscode's concern, which is why this issue never moves forward. OTOH, (b) is a much-needed feature that never gets discussed, because all feature requests that mention it are marked as a duplicate of this issue and closed (e.g. #188682).

**Comment by Tyriar** [comment]
@zocky this (a) has been on my short list of things I'd really like to do for a while. B is tracked in the highest voted issue on the repo.

If we closed this, a new issue would come up and it would take years to get the votes again...? There's clarification as to what this issue it in the top comment.

**Comment by sunglim** [comment]
@Tyriar  감사합니다 잘하셨어요 ㅋㅋ

**Comment by Tyriar** [comment]
@sunglim 오랜만이에요! 👋

**Comment by Chrislann1974** [comment]
Y

**Comment by Chrislann1974** [comment]
O

**Comment by wojpawlik** [comment]
We can now open a terminal as an edutor tab, and [move editors into floating windows](https://github.com/microsoft/vscode/issues/10121#issuecomment-1790316086).

**Comment by starball5** [comment]
> We can now open a terminal as an edutor tab, and [move editors into floating windows](https://github.com/microsoft/vscode/issues/10121#issuecomment-1790316086).

yep. already wrote this up in [my answer to the linked Stack Overflow question](https://stackoverflow.com/a/76214341/11107541) along with keyboard shortcuts to help with it.

**Comment by Upset2** [comment]
> [`](lpatch1)

**Comment by Brunojoe1995** [comment]
~~https://github.com/microsoft/vscode/issues/203705 ~~

**Comment by its2badwespend** [comment]
> "Feature" request.
> 
> The integrated terminal grows in functionality, to the point that it could soon become very useful standalone, as its *own app*. I'm only a casual VScode user (using vim most of the time), but I would definitely be interested in a well-behaved, consistent, reactively-configurable, multi-platform terminal not tied to the editor.
> 
> Obviously if such a situation were to unfold, the shared component (not entirely dissimilar to VTE) should be jointly developed and shared with VScode. Both could be able to leverage a common configuration so that whether a terminal leveraging such a theoretical common component is spawned within VScode or standalone, they behave the same.
> 
> ---
> 
> Edit from @Tyriar:
> 
> There is some confusion about the scope of this issue, here is the clarification as the thread is long:
> 
> - This issue is tracking maki__ng the terminal "stand alone", ie. being able to launch `code --term` for example (see https://github.com/microsoft/vscode/issues/34442#issuecomment-901959244 for details)
> - #10121 tracks being able to pull the terminal _view_/_panel_ out of the window
> - #10546 tracks adding tabs to the terminal, combined with #10121 this would ideally allow terminal _tabs_ to be pulled into a separate window

**Comment by ns-sjorgedeaguiar** [comment]
Now that we have Copilot in the built-in terminal, I see one more reason to get this feature. It would be awesome to be able to launch a (perhaps lightweight) terminal-only window and have all Copilot's goodies available.
